### PR TITLE
feat: add Autoclass v2.1 support

### PR DESCRIPTION
**Pull request by hemanshv**
This add support for autoclass v2.1, it will resolve https://github.com/GoogleCloudPlatform/dotnet-docs-samples/issues/2341

### CODE REVIEW

File: storage/api/Storage.Samples/GetAutoclass.cs
Comment (cojenco) [inline storage/api/Storage.Samples/GetAutoclass.cs:27]: I may not be reading this corretly, but I'd expect `bucket.Autoclass.ToggleTime` to remain the same. The only two new fields are`terminalStorageClass` and `terminalStorageClassUpdateTime`

**Review by cojenco — COMMENTED** [review]
Thanks for working on this! Generally looks good to me, just have a question on ToggleTime

File: storage/api/Storage.Samples/GetAutoclass.cs
Comment (hemanshv) [reply storage/api/Storage.Samples/GetAutoclass.cs:27]: Field `bucket.autoclass.ToggleTime` is marked as obsolete in the latest version on `google.apis.storage.v1` 
ref: https://github.com/googleapis/google-api-dotnet-client/blob/main/Src/Generated/Google.Apis.Storage.v1/Google.Apis.Storage.v1.cs#L8334

**Review by Rishabh-V — COMMENTED** [review]
Looks good. Just a couple of questions

File: storage/api/Storage.Samples/SetAutoclass.cs
Comment (hemanshv) [reply storage/api/Storage.Samples/SetAutoclass.cs:28]: yes Rishabh to make it more user friendly we can add an enum in the manual layer.

@amanda-tarafa  @cojenco Please share your thoughts on the same.

File: storage/api/Storage.Samples/GetAutoclass.cs
Comment (amanda-tarafa) [reply storage/api/Storage.Samples/GetAutoclass.cs:27]: Yes, this is a change we made to the libraries for DateTime typed fields, for better representation of data-time values. It's unrelated to autoclass, but if we are here, it is better to use the recommended new property for `ToggleTimeDateTimeOffset`.

File: storage/api/Storage.Samples/SetAutoclass.cs
Comment (amanda-tarafa) [reply storage/api/Storage.Samples/SetAutoclass.cs:28]: We have several similar other cases in Storage where we haven't added enums, and I'd prefer to not do that here. As is, it allows us to expose the AutoclassData type directly, if we add enums then we also need to add a way for users to assign the enums directly etc.

What we could do if this becomes problematic is add constants, but let's wait till we know it is needed. For the other similar cases, users have been dealing with the literal strings, no issues at all.

**Review by amanda-tarafa — COMMENTED** [review]
Looks good, I'll LGTM once we have green tests.

File: storage/api/Storage.Samples/SetAutoclass.cs
Comment (hemanshv) [reply storage/api/Storage.Samples/SetAutoclass.cs:28]: Ok Amanda that make sense Thanks

Comment (cojenco) [reply storage/api/Storage.Samples/SetAutoclass.cs:28]: I agree that we can avoid enums here. I think it's fine to use literal strings and be consistent with how the client deals with storageClass, thanks!

File: storage/api/Storage.Samples/GetAutoclass.cs
Comment (cojenco) [reply storage/api/Storage.Samples/GetAutoclass.cs:27]: Ah got it, thank you both for the clarification

**Comment by cojenco** [comment]
The feature rollout is 100% complete and is ready for merge/release. The tests should pass now, thanks @hemanshv

**Comment by amanda-tarafa** [comment]
We'll look into releasing Google.Cloud.Storage.V1 with a newer Google.Apis.Storage.v1 dependency.

**Comment by amanda-tarafa** [comment]
@hemanshv Google.Cloud.Storage.V1 [has been released](https://www.nuget.org/packages/Google.Cloud.Storage.V1/4.7.0) with the newest dependency of Google.Apis.Storage.v1.

You can wait for the renovate PR to update the dependency and then, once that's merged, rebase this PR on main.

**Comment by amanda-tarafa** [comment]
@hemanshv Storage dependencies have been updated through the repo by renovate. You can rebase this PR on main and tests should pass after. Thanks.

**Review by amanda-tarafa — APPROVED** [review]

**Pull request merged**
